controllers/replicationdestination_controller.go

```diff
@@ -93,36 +103,40 @@ func (r *ReplicationDestinationReconciler) Reconcile(ctx context.Context, req ct
 		return result, err
 	}
 
-	if r.countReplicationMethods(inst, logger) > 1 {
-		err = fmt.Errorf("only a single replication method can be provided")
-		return result, err
-	}
-	result, err = reconcileDestUsingCatalog(ctx, inst, r, logger)
-	if errors.Is(err, errNoMoverFound) { // do the old stuff
-		if inst.Spec.External != nil {
-			// Not an internal method... we're done.
-			return ctrl.Result{}, nil
-		}
-		err = fmt.Errorf("a replication method must be specified")
-	}
+	rdm, err := newRDMachine(inst, r.Client, logger, record.NewEventRecorderAdapter(r.EventRecorder))
 
-	// Set reconcile status condition
-	if err == nil {
+	// Using only external method
+	if errors.Is(err, mover.ErrNoMoverFound) && inst.Spec.External != nil {
+		return ctrl.Result{}, nil
+	}
+	// Both internal and external methods defined
```

Comment:
These checks for external methods, movers etc - could they be moved into the newRDMachine or some sort of sm.Validate() call?  I think the checks seem common to replicationsource & dest with multiple methods/external etc.

Reply:
I agree that it's redundant across RS/RD, but I'm coming up empty on how to fix it. I could use some suggestions:

- I went pretty far down the path of creating a "null" mover that registers for the external method, but the state machine still runs and updates timestamps, so that's a non-starter.
- Moving the check into newR?Machine really just moves it and doesn't reduce the duplication, best I can tell.

The unfortunate situation is that "external" is special since it doesn't use the state machine, and the logic for matching (Catalog) also doesn't seem to be re-usable.

Reply:
sorry to send you down a rabbit-hole there - sometimes these things look easier from the outside than they really are...